app/dataviz/views.py

```diff
@@ -0,0 +1,389 @@
+from django.template.response import TemplateResponse
+from django.utils import timezone
+
+from chartit import Chart, DataPool
+from dashboard.models import Profile, UserAction
+from marketing.models import EmailEvent, EmailSubscriber, GithubEvent, SlackPresence, SlackUser, Stat
+
+
+
+def filter_types(types, _filters):
+    return_me = []
+    for t in types:
+        add = False
+        for f in _filters:
+            if f in t:
+                add = True
+        if add:
+            return_me.append(t)
+
+    return return_me
+
+
+@staff_member_required
+def stats(request):
+    # get param
+    _filter = request.GET.get('filter')
+    rollup = request.GET.get('rollup')
+    _format = request.GET.get('format', 'chart')
+
+    # types
+    types = list(Stat.objects.distinct('key').values_list('key', flat=True))
+    types.sort()
+
+    # filters
+    if _filter == 'Activity':
+        _filters = [
+            'tip',
+            'bount'
+        ]
+        types = filter_types(types, _filters)
+    if _filter == 'Marketing':
+        _filters = [
+            'slack',
+            'email',
+            'whitepaper',
+            'twitter'
+        ]
+        types = filter_types(types, _filters)
+    if _filter == 'KPI':
+        _filters = [
+            'browser_ext_chrome',
+            'medium_subscribers',
+            'github_stargazers_count',
+            'slack_users',
+            'email_subscribers_active',
+            'bounties_open',
+            'bounties_ful',
+            'joe_dominance_index_30_count',
+            'joe_dominance_index_30_value',
+            'turnaround_time_hours_30_days_back',
+            'tips',
+            'twitter',
+            'user_action_Login',
+            'bounties_hourly_rate_inusd_last_24_hours',
+        ]
+        types = filter_types(types, _filters)
+
+    # params
+    params = {
+        'format': _format,
+        'types': types,
+        'chart_list': [],
+        'filter_params': f"?filter={_filter}&format={_format}&rollup={rollup}",
+        'tables': {},
+    }
+
+    for t in types:
+
+        # get data
+        source = Stat.objects.filter(key=t)
+        if rollup == 'daily':
+            source = source.filter(created_on__hour=1)
+            source = source.filter(created_on__gt=(timezone.now() - timezone.timedelta(days=30)))
+        elif rollup == 'weekly':
+            source = source.filter(created_on__hour=1, created_on__week_day=1)
+            source = source.filter(created_on__gt=(timezone.now() - timezone.timedelta(days=30*3)))
+        else:
+            source = source.filter(created_on__gt=(timezone.now() - timezone.timedelta(days=2)))
+
+        if source.count():
+            # tables
+            params['tables'][t] = source
+
+            # charts
+            # compute avg
+            total = 0
+            count = source.count() - 1
+            avg = "NA"
+            if count > 1:
+                for i in range(0, count):
```

Comment:
We can probably use `enumerate()` for some of the range iterations.

Reply:
ncie.. i didnt know about enumerate